### PR TITLE
SUBMARINE-247. Add test cases for Python Interpreter

### PR DESCRIPTION
**Pull request by luzhonghao**
### What is this PR for?
The PR aims to add some test cases for the  submarine python interpreter , including
1.PythonInterpreter::calcOnePlusOne
2.PythonInterpreter::close()
3.PythonInterpreter::cancel()

### What type of PR is it?
 Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-247

### How should this be tested?
* CI  https://travis-ci.org/luzhonghao/hadoop-submarine/builds/599067380

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by luzhonghao** [comment]
@liuxunorg  Please help me review the code , thanks

File: submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java [hunk 1]
```diff
@@ -24,35 +24,99 @@
...
     LOG.info("result = {}", result);
 
     assertEquals(result.code(), InterpreterResult.Code.SUCCESS);
     assertEquals(result.message().get(0).getData(), "2\n"); // 1 + 1 = 2 + '\n'
   }
+
+  private class infinityPythonJobforCancel implements Runnable {
+    @Override
+    public void run() {
+      String code = "import time\nwhile True:\n  time.sleep(1)";
```
Comment (liuxunorg) [inline submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java]: Why is `time.sleep(1)`?

Comment (luzhonghao) [reply submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java]: Thanks for the review, the python code is " while True: time.sleep(1)" which aims to simulate a long running code.

**Comment by liuxunorg** [comment]
LGTM

**Comment by liuxunorg** [comment]
@luzhonghao Thank you contribution test case to the Python Interpreter.
Will merge if no more comment.

File: submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java [hunk 2]
```diff
@@ -24,35 +24,98 @@
...
-  private static PythonInterpreter pythonInterpreter = null;
+  private static PythonInterpreter testPythonInterpreter1 = null;
+
+  private static PythonInterpreter testPythonInterpreter2 = null;
 
   @BeforeClass
   public static void setUp() {
-    pythonInterpreter = new PythonInterpreter();
-    pythonInterpreter.open();
+    testPythonInterpreter1 = new PythonInterpreter();
```
[the remark this answers is not in the document]
Comment (luzhonghao) [reply submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java]: Thanks for the review , the reason is that the python process is interrupted when we call cancel()  , so we have to create another python interpreter to test the close() method. Maybe we should rename the testPythonInterpreter1 and testPythonInterpreter2 for better understanding.

File: submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java [hunk 3]
```diff
@@ -24,35 +24,98 @@
...
   @Test
-  public void calcOnePlusOne() throws IOException {
+  public void calcOnePlusOne() {
     String code = "1+1";
-    InterpreterResult result = pythonInterpreter.interpret(code);
+    InterpreterResult result = testPythonInterpreter1.interpret(code);
     LOG.info("result = {}", result);
 
     assertEquals(result.code(), InterpreterResult.Code.SUCCESS);
     assertEquals(result.message().get(0).getData(), "2\n"); // 1 + 1 = 2 + '\n'
```
[the remark this answers is not in the document]
Comment (luzhonghao) [reply submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java]: Done, many thanks

File: submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuanzac) [reply submarine-workbench/interpreter/python-interpreter/src/test/java/org/apache/submarine/interpreter/PythonInterpreterTest.java]: It makes sense to me.
Thanks

**Pull request closed without merging**